crates/wasi-common/src/sys/unix/linux/filetime.rs

```diff
@@ -0,0 +1,53 @@
+use super::super::filetime::FileTime;
+use std::fs::File;
+use std::io;
+use std::sync::atomic::{AtomicBool, Ordering::SeqCst};
+
+pub(crate) const UTIME_NOW: i64 = libc::UTIME_NOW;
+pub(crate) const UTIME_OMIT: i64 = libc::UTIME_OMIT;
+
+pub(crate) fn utimensat(
+    dirfd: &File,
+    path: &str,
+    atime: FileTime,
+    mtime: FileTime,
+    symlink: bool,
+) -> io::Result<()> {
+    use super::super::filetime::{to_timespec, utimesat};
+    use std::ffi::CString;
+    use std::os::unix::prelude::*;
+
+    let flags = if symlink {
+        libc::AT_SYMLINK_NOFOLLOW
+    } else {
+        0
+    };
+
+    // Attempt to use the `utimensat` syscall, but if it's not supported by the
+    // current kernel then fall back to an older syscall.
+    static INVALID: AtomicBool = AtomicBool::new(false);
+    if !INVALID.load(SeqCst) {
```

Comment:
Could this use `Relaxed` instead of `SeqCst`, since ordering isn't particularly interesting for this variable?

Reply:
Hmm, I was trying to figure out if there could be a problem with two competing threads trying to call this bit of code simultaneously, but I cannot see any big problem except for perhaps the two threads performing a potentially failing `libc::syscall` call twice. @alexcrichton since you're the main mastermind behind this, any thoughts? :-)

Reply:
I personally automatically use `SeqCst` everywhere until I get to a point where I get benchmark number showing that it needs further optimization, and then I go and think about relaxing things. I find using `SeqCst` means that I basically don't have to think about orderings at all, which 99% of the time is what I want to do.

That's just my own personal thinking, though, and yes `Relaxed` here would be fine.